### PR TITLE
feat(wmg): Hide marker genes with marker score <0.5

### PR DESCRIPTION
**Pull request by ashin-czi**
## Reason for Change

- #4228 
- If the reason for this PR's code changes are not clear in the issue, state value/impact

## Changes

- add
    - New view in FMG panel for when no reliable marker genes are found
    - removing genes <0.5 from marker genes panel
    - limiting marker genes shown to top 25 genes
- remove
- modify

## Testing steps

rdev: https://hide-marker-genes-frontend.rdev.single-cell.czi.technology/gene-expression

### CODE REVIEW

File: frontend/src/views/WheresMyGene/common/constants.ts [hunk 1]
```diff
@@ -1 +1,3 @@
 export const HEATMAP_CONTAINER_ID = "heatmap-container-id";
+export const FMG_GENE_COUNT_LIMIT = 25;
```
Comment (atarashansky) [inline frontend/src/views/WheresMyGene/common/constants.ts:2]: Good thinking, but the gene count limit is determined as a parameter sent via the request body to the marker genes endpoint. So this is not necessary and duplicating logic.

TL;DR - you can get rid of this and don't need to `.slice(0,FMG_GENE_COUNT_LIMIT)`

File: frontend/src/common/queries/wheresMyGene.ts [hunk 2]
```diff
@@ -825,6 +828,12 @@ export function useMarkerGenes({
     return result;
   }, [data]);
 
+  function filterMarkerGenes(markerGenes: MarkerGene[]): MarkerGene[] {
+    return markerGenes
+      .filter((markerGene) => markerGene.effect_size >= FMG_GENE_STRENGTH_THRESHOLD)
+      .slice(0, FMG_GENE_COUNT_LIMIT);
```
Comment (atarashansky) [inline frontend/src/common/queries/wheresMyGene.ts:834]: see previous comment - you can safely get rid of this <3

**Review by atarashansky — CHANGES_REQUESTED** [review]
Great job, @ashin-czi! Looks good. Just one comment about the gene count limit constant you added - we can safely get rid of that and all references to it.

**Comment by ashin-czi** [comment]
Waiting on analytics: https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/gh/chanzuckerberg/single-cell-data-portal/4259

**Comment by tihuan** [comment]
@ashin-czi is this PR ready for review? Just wanted to see if the double event thing was resolved yesterday 🤞 Thank you!

**Comment by ashin-czi** [comment]
@tihuan Yes it is!

File: frontend/src/common/queries/wheresMyGene.ts [hunk 3]
```diff
@@ -796,7 +799,7 @@ export function useMarkerGenes({
   organismID,
   tissueID,
   test,
-}: FetchMarkerGeneParams): UseQueryResult<MarkerGeneResponse> {
+}: FetchMarkerGeneParams): UseQueryResult<MarkerGeneResponse<MarkerGene>> {
```
Comment (atarashansky) [inline frontend/src/common/queries/wheresMyGene.ts:802]: just curious - why was parametrizing `MarkerGene` like this necessary?

File: frontend/src/common/queries/wheresMyGene.ts [hunk 4]
```diff
@@ -834,7 +843,7 @@ export function useMarkerGenes({
             }
             return newEntries;
           },
-          [] as [string, { effect_size: number; p_value: number }][]
+          [] as [string, Partial<MarkerGene>][]
```
Comment (atarashansky) [inline frontend/src/common/queries/wheresMyGene.ts:846]: TIL `Partial` is a thing. This answers my previous question.

**Review by atarashansky — APPROVED** [review]
looks great!
